rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -132,17 +137,203 @@ void Player::play_messages_from_queue()
   } while (!is_storage_completely_loaded() && rclcpp::ok());
 }
 
+unsigned long alignment(unsigned long data_size, unsigned long last_data_size, unsigned long current_position)
+{
+  return data_size > last_data_size ? (data_size - current_position % data_size) & (data_size-1):0;
+}
+
+void Player::deal_with_string(const uint8_t *dds_buffer, bool is_wstring)
+{
+  uint32_t length;
+  size_t string_header = sizeof(uint32_t);
+
+  unsigned long one_offset = alignment(string_header, last_data_size_, current_position_);
+  current_position_ = current_position_ + one_offset;
+  memcpy(&length, (dds_buffer + current_position_ + 4), string_header);
+
+  if(!is_wstring)
+    {
+      last_data_size_ = sizeof(char);
+      current_position_ += (string_header + length);
+    }
+  else {
+      last_data_size_ = sizeof(uint32_t);
+      current_position_ += (string_header + length * sizeof(uint32_t));
+  }
+}
+
+void Player::calculate_position_with_align(const uint8_t * dds_buffer_ptr, const rosidl_typesupport_introspection_cpp::MessageMember *message_member, unsigned long stop_index)
+{
+  unsigned long one_offset = 0;
+  unsigned long data_size = 0;
+
+  for (unsigned int i = 0; i < stop_index; i++) {
+    bool is_string = false;
+    bool is_wstring = false;
+    bool is_ros_msg_type = false;
+    const rosidl_typesupport_introspection_cpp::MessageMembers * sub_members;
+    switch (message_member[i].type_id_) {
+      case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BOOL:
+        data_size = sizeof(bool);
+        break;
+      case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BYTE:
+      case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT8:
+        data_size = sizeof (uint8_t);
+        break;
+      case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_CHAR:
+      case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT8:
+        data_size = sizeof (char);
+        break;
+      case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT:
+        data_size = sizeof (float);
+        break;
+      case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT64:
+        data_size = sizeof (double);
```

Comment:
`sizeof(double)` may be sizes different than 64bits. Is this intentional?
I would have expected it to be 8 considering the switch case is `FLOAT64`

Reply:
https://github.com/ros2/rmw_fastrtps/blob/9438cca2a6a21b2436684607fea8a78624363f80/rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp#L525

:+1: 
Is an example for that:

we should replicate what was done here @scoopySHI

Reply:
https://github.com/ros2/rmw_fastrtps/blob/9438cca2a6a21b2436684607fea8a78624363f80/rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp#L525

:+1: We should replicate what was done here